lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,29 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @building = building
     end
 
     def call
-      raise NotImprementedYet
+      calc_siege_ability
+    end
+
+    private
+
+    attr_reader :building
+
+    def calc_siege_ability
+      return 0 if @building.warriors.count.zero?
+
+      day_req = 10
+      @building.warriors.each do |warrior|
+        day_req += if warrior.type == 'Warriors::Hussar'
```

Comment:
Ładniej by to chyba wyglądało, jeśli Warrior miałby zaimplementowaną metodę/atrybut np. supply_burn_rate, który zwraca 1 jeśli warrior nie ma konia i 2 jeśli go ma, wtedy nie trzeba by myśleć nad takimi ifami w tym serwisie :)